File: java/src/main/java/factory_method/product/UltraBus.java
package factory_method.product;

import factory_method.product.part.Engine;
import factory_method.product.part.Tire;

public class UltraBus extends Vehicle {
    public static final String NAME = "Ultra Bus";

    public UltraBus() {
        name = NAME;
        engine = Engine.GASOLINE;
        tire = Tire.NORMAL;
    }
}

File: src/main/kotlin/factory_method/product/UltraBus.kt
package factory_method.product

import factory_method.product.part.Engine
import factory_method.product.part.Tire

class UltraBus : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.GASOLINE
    override var tire: Tire = Tire.NORMAL

    companion object {
        const val NAME: String = "Ultra Bus"
    }
}